processing/src/main/java/org/apache/druid/query/aggregation/any/StringAnyVectorAggregator.java

```diff
@@ -78,7 +84,23 @@ public void aggregate(ByteBuffer buf, int position, int startRow, int endRow)
         if (startRow < rows.length) {
           IndexedInts row = rows[startRow];
           @Nullable
-          String foundValue = row.size() == 0 ? null : multiValueSelector.lookupName(row.get(0));
+          String foundValue;
+          if (row.size() == 0) {
+            foundValue = null;
+          } else if (aggregateMultipleValues) {
+            if (row.size() > 1) {
+              List<String> arrayList = new ArrayList<>();
+              row.forEach(rowIndex -> {
+                String chopped = StringUtils.fastLooseChop(multiValueSelector.lookupName(rowIndex), maxStringBytes);
```

Comment:
i don't think you need to call `fastLooseChop` here because `putValue` uses `StringUtils.toUtf8WithLimit` which achieves the same thing

Reply:
I removed it. Applied to arrays initially to avoid partially chopped result like `[first, seco`. applying chop at last leaves the result in weird open ended array. I think it might be expected as user can change it with increasing max bytes.

Reply:
that's fine i think, i don't really like the silent truncation at all personally; i think we should probably change first/last/any to all optionally fail instead of silently truncating results, but that doesn't need to changed now.